omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ObsResource1_8.java

```diff
@@ -315,9 +316,27 @@ public static void setConcept(Obs obs, Object value) {
 	public static void setValue(Obs obs, Object value) throws ParseException, ConversionException, IOException {
 		if (value != null) {
 			if (obs.isComplex()) {
-				byte[] bytes = DatatypeConverter.parseBase64Binary(value.toString());
+				
+				String stringValue = value.toString();
+				String[] uriParams = stringValue.split(",");
```

Comment:
@rkorytkowski 
URI data scheme is ` data:[<media type>][;base64],<data>` ([Source](https://tools.ietf.org/html/rfc2397))
`media type` (MIME Type) value cannot contain commas ([Source](https://tools.ietf.org/html/rfc2045#section-5.1), [MIME List](https://www.sitepoint.com/web-foundations/mime-types-complete-list/))
`data` value cannot contain commas (See **Base64 index table** [here](https://en.wikipedia.org/wiki/Base64))
By that, I assume that its fine to use `,` as separator, since its impossible that there are more than 1 comma in valid data URI.